marimo/_cli/convert/ipynb.py

```diff
@@ -1,13 +1,49 @@
 # Copyright 2024 Marimo. All rights reserved.
 from __future__ import annotations
 
+import ast
 import json
 
+from marimo._ast.compiler import compile_cell
+from marimo._ast.transformers import NameTransformer
 from marimo._cli.convert.utils import (
     generate_from_sources,
     load_external_file,
     markdown_to_marimo,
 )
+from marimo._runtime.dataflow import DirectedGraph
+
+
+def fixup_multiple_definitions(sources: list[str]) -> list[str]:
+    try:
+        cells = [
+            compile_cell(source, cell_id=str(i))
+            for i, source in enumerate(sources)
+        ]
+    except SyntaxError:
+        return sources
+
+    graph = DirectedGraph()
+    for cell in cells:
+        graph.register_cell(cell_id=cell.cell_id, cell=cell)
+
+    multiply_defined_names = graph.get_multiply_defined()
+    if not multiply_defined_names:
+        return sources
+
+    name_transformations = {}
+    for name in multiply_defined_names:
+        if not graph.get_referring_cells(name):
+            name_transformations[name] = (
+                "_" + name if not name.startswith("_") else name
+            )
+
+    return [
+        ast.unparse(
+            NameTransformer(name_transformations).visit(ast.parse(source))
+        )
+        for source in sources
```

Comment:
future looking, but once we have more fixes/transforms, we can chain them together